recon.py

```diff
@@ -0,0 +1,329 @@
+"""
+A Proposal for reconciliation graph and reconciliation interface.
+"""
+from abc import ABC  # Abstract Base Classes
+from typing import List, Dict
+from enum import Enum
+
+__all__ = ['MappingNode', 'EventType', 'Event', 'Cospeciation', 'Duplication', 'Transfer', 'Loss',
+           'TipTip', 'Reconciliation', 'ReconGraph']
+
+class MappingNode:
+    """
+    MappingNode is a node in the reconciliation graph that maps
+    a parasite to a host.
+    """
+    def __init__(self, parasite_vertex: str, host_vertex: str):
+        self._parasite = parasite_vertex
+        self._host = host_vertex
+
+    @property
+    def parasite(self) -> str:
+        return self._parasite
+
+    @property
+    def host(self) -> str:
+        return self._host
+    
+    def __eq__(self, other):
+        if type(self) == type(other):
```

Comment:
Be careful about using type.  It is not a thing right now, but if you had a class inheriting from MappingNode, checking types would fail.  The "pythonic" way is to use "isinstance(self, class)", which can handle inheritance.

Reply:
Will change the code to use `isinstance(self, class)`.

Reply:
Sorry, just to be clear: isinstance(obj, self.__class__), where obj is either self or other.

Reply:
I will try out the generated `__eq__` from PyCharm and see how they implement it.